Description: Don't require python3-tzdata

Author: Rebecca N. Palmer <rebecca_palmer@zoho.com>
Bug-Debian: https://bugs.debian.org/1043968
Forwarded: no

--- a/pandas/_libs/tslibs/timezones.pyx
+++ b/pandas/_libs/tslibs/timezones.pyx
@@ -65,9 +65,6 @@ cdef bint is_utc_zoneinfo(tzinfo tz):
             utc_zoneinfo = ZoneInfo("UTC")
         except zoneinfo.ZoneInfoNotFoundError:
             return False
-        # Warn if tzdata is too old, even if there is a system tzdata to alert
-        # users about the mismatch between local/system tzdata
-        import_optional_dependency("tzdata", errors="warn", min_version="2022.7")
 
     return tz is utc_zoneinfo
 
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -31,8 +31,7 @@ dependencies = [
   "numpy>=1.23.2; python_version=='3.11'",
   "numpy>=1.26.0; python_version>='3.12'",
   "python-dateutil>=2.8.2",
-  "pytz>=2020.1",
-  "tzdata>=2022.7"
+  "pytz>=2020.1"
 ]
 classifiers = [
     'Development Status :: 5 - Production/Stable',
