From: Yaroslav Halchenko <debian@onerussian.com>
Subject: BF: skip two tests which failed when xlwt was N/A

--- a/pandas/io/tests/test_excel.py
+++ b/pandas/io/tests/test_excel.py
@@ -417,6 +417,9 @@ class ReadingTestsBase(SharedItems):
 
     # GH 12292 : error when read one empty column from excel file
     def test_read_one_empty_col_no_header(self):
+        _skip_if_no_xlwt()
+        _skip_if_no_openpyxl()
+
         df = pd.DataFrame(
             [["", 1, 100],
              ["", 2, 200],
@@ -475,6 +478,9 @@ class ReadingTestsBase(SharedItems):
     def test_set_column_names_in_parameter(self):
         # GH 12870 : pass down column names associated with
         # keyword argument names
+        _skip_if_no_xlwt()
+        _skip_if_no_openpyxl()
+
         refdf = pd.DataFrame([[1, 'foo'], [2, 'bar'],
                               [3, 'baz']], columns=['a', 'b'])
 
