Description: Don't assume little-endian in test references

Author: Rebecca N. Palmer <rebecca_palmer@zoho.com>
Forwarded: no

--- a/pandas/tests/indexes/interval/test_constructors.py
+++ b/pandas/tests/indexes/interval/test_constructors.py
@@ -20,7 +20,7 @@ from pandas import (
 import pandas._testing as tm
 from pandas.core.arrays import IntervalArray
 import pandas.core.common as com
-
+from pandas.compat import is_platform_little_endian
 
 @pytest.fixture(params=[None, "foo"])
 def name(request):
@@ -41,12 +41,12 @@ class ConstructorTests:
             (Index(np.arange(-10, 11, dtype=np.int64)), np.int64),
             (Index(np.arange(10, 31, dtype=np.uint64)), np.uint64),
             (Index(np.arange(20, 30, 0.5), dtype=np.float64), np.float64),
-            (date_range("20180101", periods=10), "<M8[ns]"),
+            (date_range("20180101", periods=10), "<M8[ns]" if is_platform_little_endian() else ">M8[ns]"),
             (
                 date_range("20180101", periods=10, tz="US/Eastern"),
                 "datetime64[ns, US/Eastern]",
             ),
-            (timedelta_range("1 day", periods=10), "<m8[ns]"),
+            (timedelta_range("1 day", periods=10), "<m8[ns]" if is_platform_little_endian() else ">m8[ns]"),
         ]
     )
     def breaks_and_expected_subtype(self, request):
