Description: Xfail tests using non-functioning URLs

Author: Rebecca N. Palmer <rebecca_palmer@zoho.com>
Forwarded: no

--- pandas-0.25.3+dfsg.orig/pandas/tests/io/excel/test_readers.py
+++ pandas-0.25.3+dfsg/pandas/tests/io/excel/test_readers.py
@@ -499,6 +499,7 @@ class TestReaders:
             pd.read_excel("", engine=bad_engine)
 
     @tm.network
+    @pytest.mark.xfail(condition=True,reason="404 URL",strict=False)
     def test_read_from_http_url(self, read_ext):
         if read_ext == ".ods":  # TODO: remove once on master
             pytest.skip()
--- pandas-0.25.3+dfsg.orig/pandas/tests/io/test_html.py
+++ pandas-0.25.3+dfsg/pandas/tests/io/test_html.py
@@ -132,6 +132,7 @@ class TestReadHtml:
         assert_framelist_equal(df1, df2)
 
     @network
+    @pytest.mark.xfail(condition=True,reason="404 URL",strict=False)
     def test_spam_url(self):
         url = (
             "https://raw.githubusercontent.com/pandas-dev/pandas/master/"
