Description: Use a non-backzone timezone for test

Author: Rebecca N. Palmer <rebecca_palmer@zoho.com>
Bug-Debian: https://bugs.debian.org/1031437
Forwarded: https://github.com/pandas-dev/pandas/pull/51486

--- a/pandas/tests/tseries/offsets/test_dst.py
+++ b/pandas/tests/tseries/offsets/test_dst.py
@@ -188,7 +188,7 @@ class TestDST:
             Timestamp("1900-01-01"),
             Timestamp("1905-07-01"),
             MonthBegin(66),
-            "Africa/Kinshasa",
+            "Africa/Lagos",
             marks=pytest.mark.xfail(
                 pytz_version < Version("2020.5") or pytz_version == Version("2022.2"),
                 reason="GH#41906: pytz utc transition dates changed",
