Description: rownum is 64 bit

Author: Trupti
Bug-Debian: https://bugs.debian.org/1148010

--- a/pandas/_libs/src/parser/tokenizer.c
+++ b/pandas/_libs/src/parser/tokenizer.c
@@ -647,7 +647,7 @@ static int parser_buffer_bytes(parser_t
 static int skip_this_line(parser_t *self, int64_t rownum) {
   if (self->skipfunc != NULL) {
     PyGILState_STATE state = PyGILState_Ensure();
-    PyObject *result = PyObject_CallFunction(self->skipfunc, "i", rownum);
+    PyObject *result = PyObject_CallFunction(self->skipfunc, "L", (long long) rownum);
 
     // Error occurred. It will be processed
     // and caught at the Cython level.
