Description: Skip tests removed for copyright reasons

Author: Rebecca N. Palmer <rebecca_palmer@zoho.com>
Forwarded: no

--- a/pandas/tests/io/test_html.py
+++ b/pandas/tests/io/test_html.py
@@ -374,6 +374,7 @@ class TestReadHtml:
         assert sorted(zz) == sorted(["Repo", "What"])
 
     @pytest.mark.slow
+    @pytest.mark.skip(reason='test data removed for copyright reasons')
     def test_thousands_macau_stats(self, datapath):
         all_non_nan_table_index = -2
         macau_data = datapath("io", "data", "macau.html")
@@ -383,6 +384,7 @@ class TestReadHtml:
         assert not any(s.isna().any() for _, s in df.items())
 
     @pytest.mark.slow
+    @pytest.mark.skip(reason='test data removed for copyright reasons')
     def test_thousands_macau_index_col(self, datapath):
         all_non_nan_table_index = -2
         macau_data = datapath("io", "data", "macau.html")
@@ -565,6 +567,7 @@ class TestReadHtml:
 
         tm.assert_frame_equal(result, expected)
 
+    @pytest.mark.skip(reason='test data removed for copyright reasons')
     def test_nyse_wsj_commas_table(self, datapath):
         data = datapath("io", "data", "nyse_wsj.html")
         df = self.read_html(data, index_col=0, header=0, attrs={"class": "mdcTable"})[0]
@@ -888,6 +891,7 @@ class TestReadHtml:
         newdf = DataFrame({"datetime": raw_dates})
         tm.assert_frame_equal(newdf, res[0])
 
+    @pytest.mark.skip(reason='test data removed for copyright reasons')
     def test_computer_sales_page(self, datapath):
         data = datapath("io", "data", "computer_sales_page.html")
         msg = (
